app/views/provider/admin/cms/templates/update.js.erb

```diff
@@ -14,12 +14,14 @@ $('form .input.error').removeClass('error');
     codemirror.save()
   }
 
-  // update the preview draft link after save
+  // update the preview draft link after save, in case the path has changed
   $('#cms-preview-button a').each(function () {
     if ($(this).data('preview') === 'draft') {
       $(this).attr('href', '<%= cms_draft_url(@page) %>');
     }
   });
+  const previewButton = $('#cms-preview-button button.pf-c-dropdown__toggle-button:not(.dropdown-toggle)')
```

Comment:
The first piece that was already there (for `#cms-preview-button a`) is for the entries in the expanded dropdown, and the new piece is for the toggle button, which holds the URL in the `data-url` attribute.

I have no idea why the `published` link is not updated, I think it might be a bug, but I haven't touched it.